src/clients/FtpClient.cc

```diff
@@ -822,10 +822,12 @@ Ftp::Client::dataClosed(const CommCloseCbParams &)
 void
 Ftp::Client::writeCommand(const char *buf)
 {
-    char *ebuf;
-    /* trace FTP protocol communications at level 2 */
-    debugs(9, 2, "ftp<< " << buf);
+    debugs(9, DBG_PROTOCOL, "FTP Server REQUEST: " << ctrl.conn <<
+           "\n----------\n" <<
+           buf <<
+           "\n----------");
 
```

Comment:
If we are going to touch all these lines and add so many new ones, let's stop duplicating all this code. Please add and reuse a function. I suggest calling that new function debugProtoHeader(), debugHeader(), debugProtoMetadata().

Reply:
Done in[02047b6](https://github.com/squid-cache/squid/pull/1105/commits/02047b63724d1dbb06c16cebabe84f553b2f086b).  I made it a macro following the debugs() definition to simplify the variance of parameter types.